### PR TITLE
Add specific error page for 403 errors.

### PR DESCRIPTION
**Pull request by EricWittmann**
Instead of the generic error page, users will now see a specialty error page for 403 errors.  The error page looks like this:

image

### CODE REVIEW

File: ui/src/app/pages/basePage.tsx [hunk 1]
```diff
@@ -176,6 +183,11 @@ export abstract class PageComponent<P extends PageProps, S extends PageState> ex
         return this.state.isError ? true : false;
     }
 
+    private is403Error(): boolean {
+        return this.state.error && this.state.error.error && this.state.error.error.message &&
+            this.state.error.error.message.includes("403");
```
Comment (famartinrh) [inline ui/src/app/pages/basePage.tsx:188]: there isn't a better way to detect that the http response has status 403?

Comment (EricWittmann) [reply ui/src/app/pages/basePage.tsx:188]: I know, right?  But I couldn't find a better way.  I inspected the response we get from Axios (which is the REST client we use) and they don't seem to separately expose the HTTP response code.  At least not that I could find experimentally.

Comment (carlesarnal) [reply ui/src/app/pages/basePage.tsx:188]: If you get a response from the server (which should be the case here), it should be available using `error.response.status`.

**Review by carlesarnal — APPROVED** [review]

File: ui/src/app/pages/basePage.tsx (hunk 1 above)
Comment (EricWittmann) [reply ui/src/app/pages/basePage.tsx:188]: Did you test that in the browser?  I printed out the entire error object in the browser and there wasn't any status property.

Comment (EricWittmann) [reply ui/src/app/pages/basePage.tsx:188]: Here is an example of the entire axios error object:

```json
{
   "message": "Request failed with status code 403",
   "name": "Error",
   "stack": "Error: Request failed with status code 403\n    at createError (webpack-internal:///./node_modules/axios/lib/core/createError.js:16:15)\n    at settle (webpack-internal:///./node_modules/axios/lib/core/settle.js:17:12)\n    at XMLHttpRequest.handleLoad (webpack-internal:///./node_modules/axios/lib/adapters/xhr.js:62:7)",
   "config": {
      "url": "http://localhost:8080/t/test-registry-1/apis/registry/v2/search/artifacts?limit=10&order=asc&orderby=name",
      "method": "get",
      "headers": {
         "Accept": "application/json",
         "Authorization": "Bearer **redacted**"
      },
      "transformRequest": [
         null
      ],
      "transformResponse": [
         null
      ],
      "timeout": 0,
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
      "maxContentLength": -1,
      "maxBodyLength": -1
   }
}
```

Comment (EricWittmann) [reply ui/src/app/pages/basePage.tsx:188]: Oh snap - you are right.  There is a `response` property that gets ignored by `JSON.stringify()`.  I didn't know that was even possible.

Comment (EricWittmann) [reply ui/src/app/pages/basePage.tsx]: OK fixed.  Thanks for pointing this out.

**Pull request merged**